src/main/kotlin/decoder/Deserializer.kt

```diff
@@ -128,29 +173,34 @@ class XodusDecoder(
                     ent = checkNotNull(ent.getLink(tag)) { "should be not null $tag" }
                 ).decodeSerializableValue(des)
             }
-        }
+
     }
 
     private fun parseElement(targetSerializer: KSerializer<out Any?>, link: Entity, propertyName: String): Any? {
         if (targetSerializer.descriptor.kind == SerialKind.ENUM) {
-            return classLoader.loadClass(targetSerializer.descriptor.serialName).enumConstants[link.getProperty(propertyName) as Int]
+            return classLoader.loadClass(targetSerializer.descriptor.serialName).enumConstants[link.getProperty(
+                propertyName
+            ) as Int]
         }
         return when (targetSerializer) {
             !is GeneratedSerializer<*> -> {
-                link.getProperty(propertyName) ?: restoreObject(targetSerializer, link, propertyName)
+                link.getProperty(propertyName) ?: restoreObject(targetSerializer, link, propertyName, null)
```

Comment:
magic null, move to default value
```suggestion
                link.getProperty(propertyName) ?: restoreObject(targetSerializer, link, propertyName)
```